src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs

```diff
@@ -0,0 +1,150 @@
+        internal static DiagnosticDescriptor RuleDuplicatedBitwiseValuePart = new DiagnosticDescriptor(RuleId,
+                                                                                                       s_localizableTitle,
+                                                                                                       s_localizableMessageRuleDuplicatedBitwiseValuePart,
+                                                                                                       DiagnosticCategory.Design,
+                                                                                                       DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                                                       isEnabledByDefault: false,
+                                                                                                       helpLinkUri: null,
+                                                                                                       customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDuplicatedValue, RuleDuplicatedBitwiseValuePart);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSymbolStartAction(ssac =>
+            {
+                var enumSymbol = (INamedTypeSymbol)ssac.Symbol;
+                if (enumSymbol.TypeKind != TypeKind.Enum)
+                {
+                    return;
+                }
+
+                var invalidMembersByValue = new ConcurrentDictionary<object, List<SyntaxNode>>();
+
+                // Collect duplicated values...
+                ssac.RegisterOperationAction(oc =>
+                {
+                    var fieldInitializerOperation = (IFieldInitializerOperation)oc.Operation;
+                    if (fieldInitializerOperation.InitializedFields.Length != 1 ||
+                        !fieldInitializerOperation.Value.ConstantValue.HasValue)
+                    {
+                        return;
+                    }
+
+                    var onlyReferencesOneField = GetFilteredDescendants(fieldInitializerOperation, op => op.Kind != OperationKind.Binary)
+                        .OfType<IFieldReferenceOperation>()
+                        .Any();
+                    if (onlyReferencesOneField)
+                    {
+                        return;
+                    }
+
+                    invalidMembersByValue.AddOrUpdate(fieldInitializerOperation.Value.ConstantValue.Value,
+                        new List<SyntaxNode> { fieldInitializerOperation.Syntax.Parent },
```

Comment:
Get parent syntax to get to the field itself as we are currently on the initializer.